Kenneyjam23/Assets/Scripts/Player/WeaponBehaviour.cs

```diff
@@ -37,7 +37,10 @@ private void LateUpdate()
 
     public void OnShoot(InputAction.CallbackContext context)
     {
-        if(context.performed && _canShoot)
+        // Don't shoot a bullet if the player interacted with something in the world
+        if (GetComponent<ClickInteraction>().OnClick()) return;
```

Comment:
Better to cache this but fine in this small project.
```suggestion
        if (GetComponent<ClickInteraction>().OnClick()) return;
```